docs/changelog/123188.yaml

```
@@ -0,0 +1,14 @@
+pr: 123188
+summary: Add deprecation warning to `TransportHandshaker`
+area: Infra/Core
+type: deprecation
+issues: []
+deprecation:
+  title: Add deprecation warning to `TransportHandshaker`
+  area: REST API
+  details: >
+    Users will now see a deprecation log when a remote node, who's version is < v8.18, performs a transport handshake
+    request with nodes >= v8.18 (e.g. during a CCS/CCR request). The remote cluster will need to be upgraded to >= v8.18
+    to maintain compatibility with clusters that have been upgraded to v9.
```

Maybe something like this?

```suggestion
    Versions 9.0.0 and later of {es} will not support communication with nodes of versions earlier than 8.18.0,
    so the ability to connect to nodes of earlier versions is deprecated in this version. This applies both to
    communication within a cluster and communication across clusters (e.g. for CCS or CCR).

    {es} will report in its deprecation logs each time it opens a connection to a node that will not be supported
    from version 9.0.0 onwards. You must upgrade all your clusters to version 8.18.0 or later before upgrading
    any of your clusters to 9.0.0 or later.
```

Bonus points for turning `across clusters`, `CCS`, `CCR` and `deprecation logs` into links to the relevant docs.

---

Thank you! 

>Bonus points for turning across clusters, CCS, CCR and deprecation logs into links to the relevant docs.

How do I add a link here? (I skimmed the change logs and found no examples). Is this converted to markdown?  Will the regular [link](url) apply?

---

Bah sorry wrong syntax, I meant `<<xpack-ccr,{ccr}>>`

---

Much appreciated!